architecture/data_provider/Readme.md

```diff
@@ -0,0 +1,29 @@
+<!---
+Copyright (C) 2015, Wazuh Inc.
+Created by Wazuh, Inc. <info@wazuh.com>.
+This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+-->
+
+# Wazuh module: Data Provider architecture
+## Index
+- [Wazuh module: Data Provider architecture](#wazuh-module-data-provider-architecture)
+  - [Index](#index)
+  - [Purpose](#purpose)
+  - [Sequence diagrams](#sequence-diagrams)
+
+
+## Purpose
+Everyone knows the importance of having detailed system information from our environment to take decisions based on specific use cases. Having detailed and valuable information about our environment helps us react to under unpredictable scenarios. Wazuh agents are able to collect interesting and valuable system information regarding processes, hardware, packages, OS, network and ports.
+
+The System Inventory feature interacts with different modules to split responsibilities and optimize internal dependencies:
+- Data Provider: Module in charge of gathering system information based on OSes. This involves information about current running processes, packages/programs installed, ports being used, network adapters and OS general information.
+- DBSync: This module has one single main responsibility: Database management. It manages all database related operations like insertion, update, selection and deletion. This allows Wazuh to centralize and unify database management to make it more robust and to avoid possible data misleading.
```

Comment:
```suggestion
- DBSync: This module has one single main responsibility: Database management. It manages all database related operations like insertion, update, selection and deletion. This allows Wazuh to centralize and unify database management to make it more robust and to avoid possible misleading data.
```